pkg/validation/projects.go

```diff
@@ -58,7 +58,11 @@ func ValidateProjects(projects []v1alpha2.Project) error {
 
 		if len(gitSource.Remotes) == 0 {
 			errString += fmt.Sprintf("\nprojects %s should have at least one remote", project.Name)
-		} else if len(gitSource.Remotes) > 1 || gitSource.CheckoutFrom.Remote != "" {
+		} else if len(gitSource.Remotes) > 1 || (gitSource.CheckoutFrom != nil && gitSource.CheckoutFrom.Remote != "") {
+			if len(gitSource.Remotes) > 1 && (gitSource.CheckoutFrom == nil || (gitSource.CheckoutFrom != nil && gitSource.CheckoutFrom.Remote == "")) {
```

Comment:
This conditional sequence is kind of a nightmare, something like `if A || (B && C) then: if A && (!B || (B && !C)) then error`, and should be simplified. Maybe something like 
```go
switch len(gitSource.Remotes) {
case 0:
  errString += fmt.Sprintf("\nprojects %s should have at least one remote", project.Name)
case 1:
  if gitSource.CheckoutFrom != nil {
    if err := validateRemoteMap(gitSource.Remotes, gitSource.CheckoutFrom.Remote, project.Name); err != nil {
      errString += fmt.Sprintf("\n%s", err.Error())
    }
  }
default: // len(gitSource.Remotes) >= 2
  if gitSource.CheckoutFrom == nil || gitSource.CheckoutFrom.Remote == "" {
    errString += fmt.Sprintf("\nproject %s has more than one remote defined, but has no checkoutfrom remote defined", project.Name)
    continue
  }
  if err := validateRemoteMap(gitSource.Remotes, gitSource.CheckoutFrom.Remote, project.Name); err != nil {
    errString += fmt.Sprintf("\n%s", err.Error())
  }
}
```
Maybe some of this logic can be moved into `validateRemoteMap`? That function seems to simple to be broken out since it's just a map access.

Reply:
Thanks for the suggestion. Using a case switch does make the code more readable. 
I have also updated the starterProject validation to use case switch as well to be consistent.